### PR TITLE
Add NuGet packages page

### LINKED ISSUE #740: Document which packages are required
For example, how do people know that the Provider.Design packages are no longer used?

See https://github.com/aspnet/EntityFrameworkCore/issues/12171

### LINKED ISSUE #2305: List all EF packages to the docs
We have started putting them in announcements, but it would be good to have the full list in the docs.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #2305
Fixes #740

### CODE REVIEW

File: entity-framework/toc.yml [hunk 1]
```diff
@@ -45,6 +45,8 @@
       items:
       - name: Current and planned releases
         href: core/what-is-new/index.md
+      - name: NuGet packages
+        href: core/what-is-new/nuget-packages.md
```
Comment (AndriySvyryd) [inline entity-framework/toc.yml:49]: I'd put this under "Getting started"

**Review by AndriySvyryd — APPROVED** [review]

File: entity-framework/core/what-is-new/nuget-packages.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+- [Microsoft.EntityFrameworkCore.Tools](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Tools/) for PowerShell tooling that works in the Visual Studio [Package Manager Console](/nuget/consume-packages/install-use-packages-powershell)
+- [dotnet-ef](https://www.nuget.org/packages/dotnet-ef/) and [Microsoft.EntityFrameworkCore.Design](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Design/) for cross-platform command line tooling
+
+See [Entity Framework Core Tools Reference](xref:core/cli/index) for more information on using EF Core tooling, including how to correctly install the `dotnet-ef` tool in a project or globally.
+
+> [!TIP]
+> By default, the Microsoft.EntityFrameworkCore.Design package is installed in such a way that it will not be deployed with your application. This also means that its types cannot be transitively used in other projects. Use a normal `PackageReference` in your `.csproj` file or equivalent if you need access to the types in package.
```
Comment (bricelam) [inline entity-framework/core/what-is-new/nuget-packages.md:62]: More details in [Design-time services](https://docs.microsoft.com/ef/core/cli/services)

**Review by bricelam — APPROVED** [review]

File: entity-framework/core/what-is-new/nuget-packages.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+| Spatial support for SQL Server               | [Microsoft.EntityFrameworkCore.SqlServer.NetTopologySuite](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite/) | [NetTopologySuite](https://www.nuget.org/packages/NetTopologySuite/) and [NetTopologySuite.IO.SqlServerBytes](https://www.nuget.org/packages/NetTopologySuite.IO.SqlServerBytes/)
+| Spatial support for SQLite                   | [Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite/) | [NetTopologySuite](https://www.nuget.org/packages/NetTopologySuite/) and [NetTopologySuite.IO.SpatiaLite](https://www.nuget.org/packages/NetTopologySuite.IO.SpatiaLite/)
+| Spatial support for PostgreSQL               | [Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite](https://www.nuget.org/packages/Npgsql.EntityFrameworkCore.PostgreSQL.NetTopologySuite) | [NetTopologySuite](https://www.nuget.org/packages/NetTopologySuite/) and [NetTopologySuite.IO.PostGIS](https://www.nuget.org/packages/NetTopologySuite.IO.PostGIS/) (via [Npgsql.NetTopologySuite](https://www.nuget.org/packages/Npgsql.NetTopologySuite/))
```
Comment (bricelam) [inline entity-framework/core/what-is-new/nuget-packages.md:73]: Might be worth listing Pomelo.EntityFrameworkCore.MySql.NetTopologySuite too.

Comment (ajcvickers) [reply entity-framework/core/what-is-new/nuget-packages.md:73]: @bricelam I failed to find that package! Thanks.

File: entity-framework/toc.yml (hunk 1 above)
Comment (ajcvickers) [reply entity-framework/toc.yml:49]: @JeremyLikness @dotnet/efteam Thoughts about where this page should go?

Comment (JeremyLikness) [reply entity-framework/toc.yml:49]: I like "Getting Started."

File: entity-framework/core/what-is-new/nuget-packages.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+# EF Core NuGet Packages
+
+Entity Framework Core (EF Core) is shipped as [NuGet](https://www.nuget.org/) packages. The packages needed by an application depends on:
+
+- The type of database system being used (SQL Server, SQLite, etc.)
+- The EF Core features needed
+
+The usual process for installing packages is:
+
+- Decide on a database provider and install the appropriate package (see below)
```
Comment (smitpatel) [inline entity-framework/core/what-is-new/nuget-packages.md:18]: may be hyperlink for see below?

**Pull request merged**